randovania/games/dread/exporter/patch_data_factory.py

```diff
@@ -63,13 +63,14 @@ def get_resources_for_details(detail: ExportedPickupDetails) -> list[list[dict]]
         for conditional_resource in detail.conditional_resources
     ]
 
-    if pickup.resource_lock is not None and not pickup.respects_lock and not pickup.unlocks_resource:
-        # Add the lock resource into the pickup in addition to the expansion's resources
-        assert len(resources) == 1
-        resources[0].append({
-            "item_id": get_item_id_for_item(pickup.resource_lock.locked_by),
-            "quantity": 1,
-        })
+    if resources[0][0]["item_id"] != "ITEM_NONE":
```

Comment:
I think personally I'd turn this into an early return instead of putting all the following code into another nesting level. I'm not sure if @duncathan or @henriquegemignani have a convention/preference regarding early returns or not, but it prevents lots of deep nesting.

Reply:
I prefer whatever makes code more readable. If an early return makes it more readable, then yeah.

In this case, I'd say what's most readable is detect if it's a pickup for another game. Maybe check for `not detail.other_player` in this if, and that's it.

Reply:
Oh, that's a nice flag to use. Thanks!
Updated the pr